ext/dom/php_dom.c

```diff
@@ -1540,21 +1532,24 @@ zval *dom_nodelist_read_dimension(zval *object, zval *offset, int type, zval *rv
 
 	ZVAL_LONG(&offset_copy, zval_get_long(offset));
 
-	zend_call_method_with_1_params(object, Z_OBJCE_P(object), NULL, "item", rv, &offset_copy);
+	zend_call_method_with_1_params(object, object->ce, NULL, "item", rv, &offset_copy);
 
 	return rv;
 } /* }}} end dom_nodelist_read_dimension */
 
-int dom_nodelist_has_dimension(zval *object, zval *member, int check_empty)
+static int dom_nodelist_has_dimension(zend_object *object, zval *member, int check_empty)
 {
 	zend_long offset = zval_get_long(member);
 	zval rv;
 
 	if (offset < 0) {
 		return 0;
 	} else {
-		zval *length = zend_read_property(Z_OBJCE_P(object), object, "length", sizeof("length") - 1, 0, &rv);
+		zval obj;
+		zval *length;
 
+		ZVAL_OBJ(&obj, object);
+		length = zend_read_property(object->ce, &obj, "length", sizeof("length") - 1, 0, &rv);
```

Comment:
Not sure, but we might want to change `zend_read_property` to accept a zobj too?

Reply:
May be make sense, but together with zend_update_property...(), zend_unset_property...(), some exception functions and may be others. Let do it  separately.